src/common/src/util/sort_util.rs

```diff
@@ -13,78 +13,179 @@
 // limitations under the License.
 
 use std::cmp::{Ord, Ordering};
+use std::fmt;
 use std::sync::Arc;
 
+use parse_display::Display;
 use risingwave_pb::common::{PbColumnOrder, PbDirection, PbOrderType};
 
 use crate::array::{Array, ArrayImpl, DataChunk};
+use crate::catalog::{FieldDisplay, Schema};
 use crate::error::ErrorCode::InternalError;
 use crate::error::Result;
 
-#[derive(PartialEq, Eq, Hash, Copy, Clone, Debug)]
-pub enum OrderType {
+// TODO(rc): to support `NULLS FIRST | LAST`, we may need to hide this enum, forcing developers use
+// `OrderType` instead.
+#[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Display, Default)]
+pub enum Direction {
+    #[default]
+    #[display("ASC")]
     Ascending,
+    #[display("DESC")]
     Descending,
 }
 
-impl OrderType {
-    // TODO(rc): from `PbOrderType`
-    pub fn from_protobuf(order_type: &PbDirection) -> OrderType {
+impl Direction {
+    pub fn from_protobuf(order_type: &PbDirection) -> Direction {
         match order_type {
-            PbDirection::Ascending => OrderType::Ascending,
-            PbDirection::Descending => OrderType::Descending,
+            PbDirection::Ascending => Direction::Ascending,
+            PbDirection::Descending => Direction::Descending,
             PbDirection::Unspecified => unreachable!(),
         }
     }
 
-    // TODO(rc): to `PbOrderType`
     pub fn to_protobuf(self) -> PbDirection {
         match self {
-            OrderType::Ascending => PbDirection::Ascending,
-            OrderType::Descending => PbDirection::Descending,
+            Direction::Ascending => PbDirection::Ascending,
+            Direction::Descending => PbDirection::Descending,
+        }
+    }
+}
+
+#[allow(dead_code)]
+#[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Display, Default)]
+pub enum NullsAre {
+    #[default]
+    #[display("LARGEST")]
+    Largest,
+    #[display("SMALLEST")]
+    Smallest,
+}
+
+#[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Default)]
+pub struct OrderType {
+    direction: Direction,
+    // TODO(rc): enable `NULLS FIRST | LAST`
+    // nulls_are: NullsAre,
+}
+
+impl OrderType {
+    pub fn from_protobuf(order_type: &PbOrderType) -> OrderType {
+        OrderType {
+            direction: Direction::from_protobuf(&order_type.direction()),
+        }
+    }
+
+    pub fn to_protobuf(self) -> PbOrderType {
+        PbOrderType {
+            direction: self.direction.to_protobuf() as _,
+        }
+    }
+}
+
+impl OrderType {
+    pub const fn new(direction: Direction) -> Self {
+        Self { direction }
+    }
+
+    // Create an ascending order type, with other options set to default.
```

Comment:
```suggestion
    /// Create an ascending order type, with other options set to default.
```

Let make these comments rustdoc.